vajram/extensions/sql/vajram-sql-codegen/src/main/java/com/flipkart/krystal/vajram/sql/codegen/DefaultSqlTraitVajramGenerator.java

```diff
@@ -0,0 +1,287 @@
+import com.squareup.javapoet.WildcardTypeName;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Code generator for SQL Traits that generates ComputeVajramDef implementations wrapping SQLRead or
+ * SQLWrite vajrams.
+ *
+ * <p>This generator creates vajrams for traits annotated with @SqlQuery and @SqlUpdate.
+ */
+final class DefaultSqlTraitVajramGenerator extends AbstractSqlCodeGenerator {
+
+  // Constants
+  private static final String IMPL_SUFFIX = "_sql";
+  private static final String FACETS_SUFFIX = "_Fac";
+
+  DefaultSqlTraitVajramGenerator(VajramCodeGenContext codeGenContext) {
+    super(codeGenContext);
+  }
+
+  @Override
+  public void generate() {
+    if (!isApplicable()) {
+      return;
+    }
+
+    VajramInfo vajramInfo = codeGenContext.vajramInfo();
+    TypeElement traitElement = vajramInfo.vajramClassElem();
+
+    // Determine if this is a query or update
+    String sqlQuery = extractSqlQuery(traitElement);
+    String sqlUpdate = extractSqlUpdate(traitElement);
+
+    boolean isQuery = sqlQuery != null;
+
+    if (sqlQuery == null && sqlUpdate == null) {
+      return;
+    } else if (sqlQuery != null && sqlUpdate != null) {
+      throw util.codegenUtil()
+          .errorAndThrow("Trait cannot have both @SqlQuery and @SqlUpdate", traitElement);
+    }
+
+    String sqlStatement = isQuery ? sqlQuery : sqlUpdate;
+    TypeMirror returnType = extractGenericType(traitElement);
+
+    if (returnType == null) {
+      throw util.codegenUtil()
+          .errorAndThrow("Could not extract return type from TraitRoot", traitElement);
+    }
+    TypeMirror entityType = extractEntityTypeFromList(returnType);
+    if (entityType == null) {
+      throw util.codegenUtil()
+          .errorAndThrow("Could not extract entity type from TraitRoot", traitElement);
+    }
+    validateTraitElement(traitElement, returnType, entityType, isQuery);
+    // Generate the implementation class
+    String packageName = util.codegenUtil().getPackageName(traitElement);
+    String implClassName = traitElement.getSimpleName().toString() + IMPL_SUFFIX;
+
+    TypeSpec implClass =
+        generateComputeVajram(
+            traitElement, implClassName, sqlStatement, returnType, entityType, isQuery);
+
+    // Build JavaFile with static imports
+    String facetsClassName = implClassName + FACETS_SUFFIX;
+    JavaFile javaFile =
+        JavaFile.builder(packageName, implClass)
+            .addStaticImport(IfAbsent.IfAbsentThen.class, "FAIL")
+            .addStaticImport(InputResolvers.class, "*")
+            .addStaticImport(ClassName.get(packageName, facetsClassName), "*")
+            .indent("  ")
+            .build();
+
+    try {
+      javaFile.writeTo(util.codegenUtil().processingEnv().getFiler());
```

Comment:
using `util.generateSourceFile` method to get standard code generation best practices like `@Generated` annotation etc